source/memory-store.ts

```diff
@@ -136,5 +152,72 @@ export default class MemoryStore implements Store {
 	 */
 	shutdown(): void {
 		clearInterval(this.interval)
+		void this.resetAll()
+	}
+
+	private resetClient(client: Client, now = Date.now()) {
+		client.totalHits = 0
+		client.resetTime.setTime(now + this.windowMs)
+	}
+
+	/**
+	 * Refill the pool, set previous to current, reset current
+	 */
+	private clearExpired() {
+		// At this point, all clients in previous are expired
+
+		// calculate the target pool size
+		this.recentNew.push(this.newClients)
+		if (this.recentNew.length > 10) this.recentNew.shift()
+		this.newClients = 0
+		const targetPoolSize = Math.round(average(this.recentNew))
+
+		// Calculate how many entries to potentially copy to the pool
+		let poolSpace = targetPoolSize - this.pool.length
+
+		// Fill up the pool with expired clients
+		for (const client of this.previous.values()) {
+			if (poolSpace > 0) {
+				this.pool.push(client)
```

Comment:
Should we run `resetClient` on the client when adding it to the pool? That way, the pool might take up less resources.

Reply:
Nah, it takes up the same amount of resources either way, and we have to reset the date when we pull it out of the pool, so it would just be doing extra work to also reset it when we put it in.

Reply:
Ah I didn't consider the date part, makes sense not to do it here then.